nats/src/main.rs

```diff
@@ -229,11 +286,26 @@ impl ProviderHandler for NatsMessagingProvider {
     #[instrument(level = "info", skip(self))]
     async fn delete_link(&self, actor_id: &str) {
         let mut aw = self.actors.write().await;
-        if aw.remove(actor_id).is_some() {
-            info!("nats closing connection for actor {}", actor_id);
-            // close and drop the connection
-            // dropping the client should close it
-        } // else ignore: it's already been dropped
+
+        if let Some(NatsClientBundle { sub_handles, .. }) = aw.get(actor_id) {
+            debug!(
+                "closing [{}] NATS subscriptions for actor [{}]...",
+                sub_handles.len(),
+                actor_id,
+            );
+
+            // Stop every subscribe loop related to the actor
+            for (sub, handle) in sub_handles {
+                debug!("ending subscription to subject [{}]...", sub);
+                handle.abort();
+                drop(handle);
```

Comment:
No need for the manual drop here. You can remove this. Technically it is dropping at the end of each iteration of the loop

Reply:
Sure will do -- and actually dropping *detaches* anyway. Definitely didn't need this.

- [ ] remove explicit `drop` for handle